File: C2108G2/IDP/depe01/student.dart
import 'dart:io';
class Student {
  String name;
  int age;
  double weight;
  double height;
  double gpa;
  double get bmi => weight / (height * height);
  @override
  String toString() {
    return 'name: ${name},'
        'age: ${age},'
        'weight: ${weight},'
        'height: ${height},'
        'bmi: ${bmi}';
  }
  String get range {
    if (gpa < 0 || gpa > 10) {
      throw new Exception("Value must be > 0 and <= 10");
    } else if (gpa >= 8.5) {
      return "A";
    } else if (gpa >= 7.5) {
      return "B";
    } else if (gpa >= 6.5) {
      return "C";
    } else {
      return "D";
    }
  }

  Student({
    required this.name,
    required this.age,
    required this.height,
    required this.weight,
    required this.gpa
  });

  factory Student.input() {
    print('Enter name: ');
    String _name = stdin.readLineSync() ?? '';

    print('Enter age: ');
    int _age = int.parse(stdin.readLineSync() ?? '18');

    print('Enter weight: ');
    double _weight = double.parse(stdin.readLineSync() ?? '0');

    print('Enter height: ');
    double _height = double.parse(stdin.readLineSync() ?? '0');

    print('Enter gpa: ');
    double _gpa = double.parse(stdin.readLineSync() ?? '0');

    return new Student(
        name: _name,
        age: _age,
        height: _height,
        weight: _weight,
        gpa: _gpa
    );
  }
}

File: C2407LDoiCan/baitap21062025/src/main/java/com/aptech/bai03/Student.java
package com.aptech.bai03;

import java.io.InvalidObjectException;
import java.util.Scanner;

public class Student {
    private int id;//private, protected, public... => access modifier
    private String fullName;
    private Integer age;
    private double gpa;

    public Student(int id, String fullName, Integer age, double gpa) {
        this.id = id;
        this.fullName = fullName;
        this.age = age;
        this.gpa = gpa;
    }
    public Student() {}

    public int getId() {
        return id;
    }

    public void setId(int id)  {
//        if(id < 0) {
//            throw new InvalidObjectException("id must be > 0");
//        }
        this.id = id;
    }

    public String getFullName() {
        //nguyen van a => Nguyen Van A
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
//        if(age < 0) {
//            throw new Exception("Age cannot be negative");
//        }
        this.age = age;
    }

    public double getGpa() {
        return gpa;
    }

    public void setGpa(double gpa) {
        this.gpa = gpa;
    }
    public void displayInfo() {
        System.out.println("Id: "+this.id);
        System.out.println("Fullname : "+this.getFullName());
        System.out.println("Age: "+this.age);
        System.out.println("GPA: "+this.gpa);
    }
    //factory method
    public static Student input(){
        System.out.println("Enter id:");
        int id = getScanner().nextInt();

        System.out.println("Enter fullname: ");
        String fullName = getScanner().next();

        System.out.println("Enter age: ");
        Integer age = getScanner().nextInt();

        System.out.println("Enter gpa: ");
        double gpa = getScanner().nextDouble();
        return new Student(id, fullName, age, gpa);
    }
    private static Scanner getScanner() {
        return new Scanner(System.in);
    }
}
